src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/PersistorBase.java

```diff
@@ -191,31 +191,48 @@ public synchronized void deleteOldDBFile()
       LOG.error("RCA: Caught SecurityException while trying to delete old DB file. ", e);
     }
 
-    if (!tableNames.contains(tableName)) {
-      LOG.info(
-          "RCA: Table '{}' does not exist. Creating one with columns: {}",
-          tableName,
-          flowUnit.getSqlSchema());
-      createTable(tableName, flowUnit.getSqlSchema());
-      tableNames.add(tableName);
+    try {
+      writeFlowUnit(flowUnit, tableName);
+    } catch (SQLException e) {
+      LOG.error(
+          "RCA: Caught SQLException while writing flowuni.", e);
     }
-    int lastPrimaryKey = insertRow(tableName, flowUnit.getSqlValue());
+  }
 
-    if (flowUnit.hasResourceSummary()) {
-      write(
-          flowUnit.getResourceSummary(),
-          tableName,
-          getPrimaryKeyColumnName(tableName),
-          lastPrimaryKey);
+  private synchronized <T extends ResourceFlowUnit> void writeFlowUnit(
+      T flowUnit, String tableName) throws SQLException {
+    try {
+      if (!tableNames.contains(tableName)) {
+        LOG.info(
+            "RCA: Table '{}' does not exist. Creating one with columns: {}",
+            tableName,
+            flowUnit.getSqlSchema());
+        createTable(tableName, flowUnit.getSqlSchema());
+        tableNames.add(tableName);
+      }
+      int lastPrimaryKey = insertRow(tableName, flowUnit.getSqlValue());
+
+      if (flowUnit.hasResourceSummary()) {
+        writeSummary(
+            flowUnit.getResourceSummary(),
+            tableName,
+            getPrimaryKeyColumnName(tableName),
+            lastPrimaryKey);
+      }
+    } catch (SQLException e) {
+      LOG.info(
+          "RCA: Fail to write into table '{}', try recreating the DB", tableName);
+      openNewDBFile();
```

Comment:
we are opening a new file but we are not trying to write in the new file one more time. We are throwing away this flow unit here